### ISSUE TITLE
Message is written but not sent (previously 'error: "writeMessage" Failed')

### ISSUE THREAD

**Issue by joebarnes1983** [brismuth/google-voice-bulk-texter#111]
**Describe the bug**
Texts do not send. Cycles through the first number in the list repeated before displaying error message stating error: "writeMessage" Failed

**URL**
https://voice.google.com/u/1/messages?itemId=draft-B531F607-C8C0-4574-83EB-FE461118E61C

**Expected behavior**
It was expected to send a text message to 3 recipients

**Screenshots**

**Additional context**
Extension has been working flawlessly for weeks

**Comment by jlamb504** [comment]
Thank you. I am having the same issue today.

**Comment by thiasdavid** [comment]
Same issue yesterday and today; I was persistent at it yesterday and got it to work somehow. Mostly just refreshing and trying again. But, today, no luck.

**Comment by joebarnes1983** [comment]
I have tried refreshing, reinstalling, and everything I can think of and continue to get the same error.

**Comment by brismuth** [comment]
Thanks for reporting this issue. 

It sounds like Google is probably updating the Google Voice interface. I haven't gotten the update yet, so the extension is still working for me and I'm not able to reproduce the issue or submit a fix yet. 

If you're experiencing this issue and you want to submit a fix, you can follow these instructions to install the extension from source (https://github.com/brismuth/google-voice-bulk-texter#chrome-install-from-source). 
Line 12 of this file is the one that needs to be updated: https://github.com/brismuth/google-voice-bulk-texter/blob/master/src/contentScripts/main.js#L12
You can use the Chrome element inspector to find a new working selector for the text area.

**Comment by thiasdavid** [comment]
I'm sorry; I don't know coding at all. I'm so appreciative of this code and project. I'm an educator at an online High School and this helps me to stay in contact with my students. It's making a difference. I copied the element from Voice; I don't know what I should be looking for to fix the problem. It's below.

"<div contenteditable="true" flex="noshrink" gv-focus-when="ti1" gv-placeholder="Type a message" gv-test-id="gv-message-input" id="gv-message-input" ng-blur="ctrl.onMessageEntryBlur()" ng-class="::ctrl.Css.MESSAGE_INPUT" ng-if="::ctrl.isEmojiModEnabled()" ng-keydown="ctrl.onKeydown($event)" role="textbox" class="B3EWm-PWVi0d flex-noshrink"></div>"

**Comment by brismuth** [comment]
@thiasdavid thanks for pasting that in here! I can't test my changes since I'm not affected by the Google Voice update yet, but I've pushed a possible fix in version 1.1.7.

Can you update following these instructions and see if it resolves the issue for you?
https://github.com/brismuth/google-voice-bulk-texter#update-it

**Comment by Filipe-github** [comment]
> @thiasdavid thanks for pasting that in here! I can't test my changes since I'm not affected by the Google Voice update yet, but I've pushed a possible fix in version 1.1.7.
> 
> Can you update following these instructions and see if it resolves the issue for you?
> https://github.com/brismuth/google-voice-bulk-texter#update-it

Hi, I did this today because I was having the same issue. This version gets the message and number filled in, but doesn't actually send the message. 

Thank you so much for this tool @brismuth

**Comment by brismuth** [comment]
@Filipe-github @thiasdavid can one of you copy the selector from the send button and paste that in here?

Also, can you confirm that when the message is pasted in, that the send button becomes activated/enabled?

**Comment by Filipe-github** [comment]
> @Filipe-github @thiasdavid can one of you copy the selector from the send button and paste that in here?
> 
> Also, can you confirm that when the message is pasted in, that the send button becomes activated/enabled?

Hi, send arrow turns green (from grey), but does not send. 

If you can walk me through on how to copy the selector, I'd be happy to do so. I'm not sure how to do that. I don't actually code or anything, I just made this account to respond here and help you with diagnostics. I use this tool almost every other day. Thank you!

**Comment by coachmartell20** [comment]
I'm having the same issue... Is there a fix yet?

**Comment by Filipe-github** [comment]
> @Filipe-github @thiasdavid can one of you copy the selector from the send button and paste that in here?
> 
> Also, can you confirm that when the message is pasted in, that the send button becomes activated/enabled?

is there anything I can add here?

**Comment by coachmartell20** [comment]
> > @Filipe-github @thiasdavid can one of you copy the selector from the send button and paste that in here?
> > Also, can you confirm that when the message is pasted in, that the send button becomes activated/enabled?
> 
> is there anything I can add here?

Maybe this?

        gvSendButtonOld: 'gv-icon-button[icon-name="send"] button[aria-label="Send message"]',
	gvSendButtonNew: 'gv-icon-button-ng2[icon-name="send"][label="Send message"] button',

**Comment by jackehendrix** [comment]
i have ver 1.1.7 but cant send an text.  yesterday it appeared to send only the first message (3 total) but it didn't send the other two?  today i can't get it to send any?  I know this may not be the place to post, but here it is

**Comment by coachmartell20** [comment]
> i have ver 1.1.7 but cant send an text. yesterday it appeared to send only the first message (3 total) but it didn't send the other two? today i can't get it to send any? I know this may not be the place to post, but here it is

No..... This the right place.  I'm hoping someone is able to fix.

**Comment by brismuth** [comment]
It looks like v1.1.7 introduced a small separate issue and I've resolved that with v.1.1.8 - but I am still waiting on someone to send me a copy of the "send button" HTML in order to fully support the new Google Voice update. The string of text that @thiasdavid sent earlier for the message area is exactly what I'm looking for, except for the send button instead.

@Filipe-github @thiasdavid @coachmartell20, any of you should be able to get me that HTML following these instructions hopefully:

To copy the HTML, right click on the send button, and click "Inspect".

You'll see something like the image below. Then hit the up and down arrows on your keyboard until you get to an element with the word "button" in it or something similar. In the case below, which is the previous version of Google Voice, I'd have wanted you to highlight the `gv-icon-button-ng2` element, copy it, and send me that.

![image](https://user-images.githubusercontent.com/5679695/75389416-b18ddf80-58a3-11ea-80eb-0f45f161b31a.png)

**Comment by Filipe-github** [comment]
<gv-icon-button-ng2 [disabled]="!ctrl.isAbleToSend()" ng-click="ctrl.onClickSendButton()" color="primary" gv-test-id="send_message" icon-name="send" label="Send message" tooltip-direction="before" role="button" tabindex="-1" ng-version="0.0.0-PLACEHOLDER" aria-activedescendant="ib5"><base-icon-button _nghost-obf-c2=""><button _ngcontent-obf-c2="" mat-icon-button="" class="mat-icon-button mat-button-base mat-primary" aria-label="Send message" id="ib5" tabindex="0"><span class="mat-button-wrapper"><mat-icon class="mat-icon notranslate mat-icon-no-color" role="img" aria-hidden="true"><svg xmlns="http://www.w3.org/2000/svg" width="100%" height="100%" viewBox="0 0 24 24" fit="" preserveAspectRatio="xMidYMid meet" focusable="false"><path d="M2 3v18l20-9L2 3zm2 11l9-2-9-2V6.09L17.13 12 4 17.91V14z"></path></svg></mat-icon></span><div class="mat-button-ripple mat-ripple mat-button-ripple-round" matripple=""></div><div class="mat-button-focus-overlay"></div></button></base-icon-button></gv-icon-button-ng2>

**Comment by Filipe-github** [comment]
the word button is used 248 times @brismuth  so I am not sure I got the right one. Here's a screenshot.
![Screen Shot 2020-02-26 at 7 00 49 PM](https://user-images.githubusercontent.com/61366636/75399529-718e2300-58ca-11ea-929e-c3d9c2939ea7.png)

**Comment by coachmartell20** [comment]
![image](https://user-images.githubusercontent.com/61508821/75401638-bddc6180-58d0-11ea-9bfd-4a0502944b9a.png)

Now this is the error.     It sends the 1st message.... then puts the 1st number back into the box but doesn't send.

**Comment by coachmartell20** [comment]
Here is send button

<button class="btn btn-success" id="send-messages-button" title="Pressing this will send the composed message to all of the listed contacts.">Send Messages</button>

**Comment by coachmartell20** [comment]
Is there a fix in the works?   Its been a week.   Help please!!!! @brismuth @Filipe-github

**Comment by Filipe-github** [comment]
I don't know of one @coachmartell20 . If there's anything you need from us @brismuth let us know. :)

**Comment by brismuth** [comment]
@Filipe-github that appears to be the right button - but the problem is that the code looks identical to the code from the version of Google Voice that I'm running (see my screenshot). Same element names etc. 
The extension is looking for an element that matches this selector:
`gv-icon-button-ng2[icon-name="send"][label="Send message"] button`
and from what I can tell, that should work on the HTML code you pasted in. 

If the extension isn't automatically clicking the send button for you, I'm at a loss as to how to debug and fix without reproducing the issue on my own account.

**Comment by brismuth** [comment]
@coachmartell20 sorry I haven't responded in the last week. I wasn't sure what to do as a next step to debug this or help out since the extension is still working fine on my account.

That is the nature of this extension though - it is provided free of charge, and it is completely unsupported by Google Voice. They can make breaking changes at any time and roll them out slowly to their user base, and there's nothing I can do about it except do my best to figure out how to fix it when they do. 😄

**Comment by thiasdavid** [comment]
@brismuth you're a gem and I appreciate everything you're doing!

**Comment by gmt** [comment]
I am seeing a behavior where the message is typed in, sent to the first recipient, and then it loops until timeout trying to confirmSent().  I wonder if that's what folks are referring to?

Drilling into that, a bit, I got the code in a debugger and noticed that the text strings were out of alignment relative to each other.  Not sure exactly what the "rules" are causing this, but my particular message was long, and had several sentences.  I had double spaced them, like:
```(ruler)-01234567890123456789
     this.  When...
```
those strings were compared, (https://github.com/brismuth/google-voice-bulk-texter/blob/29c381d8cab49a33a33cdb38a516edeed139b888/src/contentScripts/googleVoiceManager.js#L196) mostRecentMessage.innerText had my original text with two spaces but this.messagesToSend[this.currentNumberSending] would contain just once space in the same spot.  So, perhaps gv is now automagically translating double spaces into single spaces as a feature?  Or perhaps there is a quoting and/or input-sanitizing bug somewhere in the chain of agents on bulk-texter's end.  Frankly, I just took all the double-spaces out of my input, as by this point, I needed to get out of debugging mode and into bulk texting mode quick fast in a hurry :)

So... ?   More of a hint as to why you might be failing to repro-, than a full description of the problem... plus, if bug reporters above are being super meticulous, then, I've experienced a completely different issue from what other people are bumping up against, here, and it just happens to result in somewhat-similar failure modalities.  But if people above are normal humans, it's probably the same modality :)

**Comment by brismuth** [comment]
@gmt thanks for your comment, and for your debugging efforts. 

In this thread, users are reporting that the extension is unable to send even the first message, and it is most likely due to Google Voice rolling out a breaking update. This is not the first time Google has done this in the history of the extension, and this time matches the pattern of all the previous times (it suddenly broke for some people after working for them for a long time, and it has slowly broken for more and more people over time). 

Your issue is actually more similar to #61, and I’d encourage you to comment over on that thread. I think it’s probably a good idea to reduce the specificity of the confirmation code there to fix both issues for #61. I’ll do that when I have the chance and post on #61 when that’s done.

**Comment by gmt** [comment]
> Your issue is actually more similar to #61, and I’d encourage you to comment over on that thread

Yep.... looks like a type-before-read error, above, on my part... next time a labor monitoring drone gets its propellers inexplicably jammed with coal tar I'll endeavor to steal away from the re-education facility and spend some time looking into this further :)

**Comment by adefilip** [comment]
FYI, I went to firefox and had no problem using the extension there today. Might see if that works for you.

**Comment by thiasdavid** [comment]
Sorry I was out of the loop for awhile. 1.1.8 is working for me in Chrome now! (1.1.7 isn't working in Firefox!) And, I promise I'll be donating soon! Thank you so much!

**Comment by brismuth** [comment]
Glad it's working again! Maybe that means Google is rolling back the change they made, or maybe they made an additional change that just happened to make the extension work again.

@thiasdavid I've just rolled out 1.1.8 to the Firefox extension store, it should hopefully be automatically updated some time today.

**Comment by deanloew** [comment]
Thanks for the updates, brismuth.  Unfortunately, I have the latest version 1.2.0 on Firefox, and it refuses to send even the first text.  I have been using this for over a year, so it has been working okay.  My firefox version is latest as of today, 74.0.   It gives this error:  "Google Voice bulk texter v1.2.0:
Text failed. You can find support resources by opening the Google Voice Bulk Texter popup and clicking "Get Help" at the bottom.  Error: "fillNumberInput" failed. "    I am a amateur web developer, so if I can help, let me know.  Thanks!

**Comment by brismuth** [comment]
@deanloew #102 is the issue that’s affecting you, this is a different issue that is affecting a different subset of users. I’ve added all the info I’ve got into #102, and you’re welcome to experiment with the extensions code and try fixing it and submitting a pull request. If you use chrome, it should also resolve the issue for you unless you end up being affected by this issue (#111) as well.

**Comment by brismuth** [comment]
Closing this, the extension seems to be working fine for me in both Chrome and Firefox and I haven't seen anyone else reporting issues on this thread in a while.

If anyone on here is still affected, just comment and I'll reopen.

**Comment by brismuth** [comment]
Hey @edthefixer, glad you like the extension. This thread is closed and was for a different issue - however, your issue is addressed on the Support page:
https://github.com/brismuth/google-voice-bulk-texter/blob/master/support.md#account-suspension--error-confirmsent-failed

Good luck!
